pkg/config/v1alpha1/types.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	configv1alpha1 "k8s.io/component-base/config/v1alpha1"
+)
+
+// ControllerConfigurationSpec defines the desired state of GenericControllerConfiguration
+type ControllerConfigurationSpec struct {
+	// SyncPeriod returns the SyncPeriod
```

Comment:
Should we use the real documentation (or a brief version of the godoc) for these properties here?
Or at the very least point the reader to the actual documentation? That is, maybe `SyncPeriod returns the manager.SyncPeriod`.

That way, the documentation is somewhat useful to the reader.

Reply:
Agreed, I'll be honest I punted until we had the design ironed out a bit more before going back and updating docs so I wasn't moving it around a bunch but it seems we're nearing that time so I can go and update all these comments to reflect their actual usage.